pkg/volume/csi/csi_attacher_test.go

```diff
@@ -1630,7 +1607,30 @@ func newTestWatchPlugin(t *testing.T, fakeClient *fakeclient.Clientset, setupInf
 		t.Fatalf("cannot assert plugin to be type csiPlugin")
 	}
 
-	return csiPlug, fakeWatcher, tmpDir, fakeClient
+	attachmentWatchCreated, fakeWatcher := notifyVolumeAttachmentWatchCreated(fakeClient)
+
+	return csiPlug, attachmentWatchCreated, fakeWatcher, tmpDir, fakeClient
+}
+
+// notifyVolumeAttachmentWatchCreated returns a channel that is closed once a VolumeAttachment watch is created.
+// This is useful for ensuring a watch has been created prior to injecting events.
+func notifyVolumeAttachmentWatchCreated(client *fakeclient.Clientset) (chan core.Action, *watch.RaceFreeFakeWatcher) {
+	attachmentWatchCreated := make(chan core.Action)
+	fakeWatcher := watch.NewRaceFreeFake()
+
+	client.Fake.PrependWatchReactor("volumeattachments", func(action core.Action) (bool, watch.Interface, error) {
+		select {
+		case <-attachmentWatchCreated:
+			// Already closed.
+		default:
+			close(attachmentWatchCreated)
+		}
+		return true, fakeWatcher, nil
```

Comment:
Shouldn't this return false to allow other reactors to react normally?

Reply:
From my understanding, if we return `false` then this `fakeWatcher` will not be returned to the user.
https://github.com/kubernetes/kubernetes/blob/9406e145c25e4c3041ec82529176957f164fdb08/staging/src/k8s.io/client-go/testing/fake.go#L160-L171

Reply:
Is it necessary for "this" fakeWatcher to be returned? It still calls React() which means the attachmentWatchCreated channel will be closed. Isnt it?

Reply:
From my understanding it was necessary because of a race condition. This thread has more context: https://github.com/kubernetes/kubernetes/pull/97499#discussion_r549823785